llvm/lib/IR/AsmWriter.cpp

```diff
@@ -4161,6 +4165,11 @@ void AssemblyWriter::printFunction(const Function *F) {
     writeOperand(F->getPersonalityFn(), /*PrintType=*/true);
   }
 
+  if (PrintPerfData && F->getEntryCount()) {
```

Comment:
Why not just if getMetadata returns non-null? getEntryCount does other stuff that is being ignored here.